WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -743,6 +757,25 @@ final class EditableOrderViewModel: ObservableObject {
         })
     }
 
+    /// Saves the latest Customer Details state if the view is dismissed. Eg: on IPads, automatic modal dismissal on size class change
+    ///
+    func saveInflightCustomerDetails() {
+        let address = Address(firstName: latestAddressFormFields?.firstName ?? "",
+                              lastName: latestAddressFormFields?.lastName ?? "",
+                              company: latestAddressFormFields?.company,
+                              address1: latestAddressFormFields?.address1 ?? "",
+                              address2: latestAddressFormFields?.address2,
+                              city: latestAddressFormFields?.city ?? "",
+                              state: latestAddressFormFields?.state ?? "",
+                              postcode: latestAddressFormFields?.postcode ?? "",
+                              country: latestAddressFormFields?.country ?? "",
+                              phone: latestAddressFormFields?.phone,
+                              email: latestAddressFormFields?.email)
```

Comment:
```suggestion
        guard let latestAddressFormFields else {
            return
        }
        let address = Address(firstName: latestAddressFormFields.firstName,
                              lastName: latestAddressFormFields.lastName,
                              company: latestAddressFormFields.company,
                              address1: latestAddressFormFields.address1,
                              address2: latestAddressFormFields.address2,
                              city: latestAddressFormFields.city,
                              state: latestAddressFormFields.state,
                              postcode: latestAddressFormFields.postcode,
                              country: latestAddressFormFields.country,
                              phone: latestAddressFormFields.phone,
                              email: latestAddressFormFields.email)
```

This might be a _start_ for preventing unnecessary syncs, i.e. when the address hasn't changed, we don't do anything.

It won't fix that fully, I don't think, because changes which have been made and synced would get synced again...

Reply:
Sounds good, I've added your changes along with a check to the latest order synchronizer state on https://github.com/woocommerce/woocommerce-ios/pull/12306/commits/20ef604e2c024e7fd800af27c85eec3887857707